### PR TITLE
Put LiveKit info into the `m.call` state event

### PR DESCRIPTION
**Pull request by SimonBrandner**
Fixes https://github.com/vector-im/voip-internal/issues/168
Requires https://github.com/vector-im/element-call/pull/1159

---
This PR currently has none of the required changelog labels.

Add one of: `T-Deprecation`, `T-Enhancement`, `T-Defect`, `T-Task` to indicate what type of change this is plus `X-Breaking-Change` if it's a breaking change.

### CODE REVIEW

**Comment by dbkr** [comment]
We could consider holding off on this until the openID stuff lands so we can just advertise the jwt service URL and have that return the sfu url, to minimise the number of different things in the wild.

**Comment by SimonBrandner** [comment]
> We could consider holding off on this until the openID stuff lands so we can just advertise the jwt service URL and have that return the sfu url, to minimise the number of different things in the wild.

Sounds alright!

**Review by robintown — APPROVED** [review]

File: src/webrtc/groupCall.ts [hunk 1]
```diff
@@ -324,6 +329,8 @@ export class GroupCall extends TypedEventEmitter<
         this.client.groupCallEventHandler!.groupCalls.set(this.room.roomId, this);
         this.client.emit(GroupCallEventHandlerEvent.Outgoing, this);
 
+        const focus = this._foci[0];
```
[the remark this answers is not in the document]
Comment (t3chguy) [reply src/webrtc/groupCall.ts:332]: Don't think we can use `Array::at` yet - our babel config doesn't transpile it into something more widely supported https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at#browser_compatibility

[the remark this answers is not in the document]
Comment (robintown) [reply src/webrtc/groupCall.ts:332]: Oh well :(

[the remark this answers is not in the document]
Comment (t3chguy) [reply src/webrtc/groupCall.ts:332]: I think type-wise they're identical also. If the array is empty or the index is out of bounds it'll yield `undefined`, same as `Array::at` - the latter has a nice party trick of correctly handling negative indices though!

[the remark this answers is not in the document]
Comment (robintown) [reply src/webrtc/groupCall.ts:332]: You'd think, but I'm pretty sure TypeScript puts different types on them, making `array[i]` be a `T` while `array.at(0)` is a `T | undefined` - in other words, `array[i]` contains a type assertion that the array is long enough for that index to exist

[the remark this answers is not in the document]
Comment (t3chguy) [reply src/webrtc/groupCall.ts:332]: Interesting. I can reproduce. I could have sworn TS doing the right thing for non-non empty arrays

**Pull request merged**